File: app/src/main/java/com/example/services/push/InstanceIdService.java
package com.example.services.push;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;

import android.support.annotation.NonNull;

import com.example.android.utils.LogUtility;

public class InstanceIdService extends FirebaseInstanceIdService {
  @Override
  public void onTokenRefresh() {
    String refreshedToken = FirebaseInstanceId.getInstance().getToken();

    if (refreshedToken == null || refreshedToken.isEmpty()) {
      return;
    }

    sendRegistrationToServer(refreshedToken);
  }

  private void sendRegistrationToServer(@NonNull String refreshedToken) {
    LogUtility.d("( FIREBASE -> onTokenRefresh ) ", refreshedToken);
  }
}

File: app/src/main/kotlin/com/example/services/push/InstanceIdService.kt
package com.example.services.push

import com.example.utils.LogUtility
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService

class InstanceIdService : FirebaseInstanceIdService() {
  override fun onTokenRefresh() {
    val refreshedToken = FirebaseInstanceId.getInstance().token

    if (refreshedToken == null || refreshedToken.isEmpty()) {
      return
    }

    sendRegistrationToServer(refreshedToken)
  }

  private fun sendRegistrationToServer(refreshedToken: String) {
    LogUtility.d("( FIREBASE -> onTokenRefresh ) ", refreshedToken)
  }
}
